src/jit/hwintrinsicxarch.cpp

```diff
@@ -265,6 +265,39 @@ bool Compiler::compSupportsHWIntrinsic(InstructionSet isa)
                                                     isFullyImplmentedISAClass(isa));
 }
 
+//------------------------------------------------------------------------
+// impUnsupportedHWIntrinsic: returns a node for an unsupported HWIntrinsic
+//
+// Arguments:
+//    method     -- method handle of the intrinsic function.
+//    sig        -- signature of the intrinsic call
+//    mustExpand -- true if the intrinsic must return a GenTree*; otherwise, false
+//
+// Return Value:
+//    a gtNewMustThrowException if mustExpand is true; otherwise, nullptr
+//
+GenTree* Compiler::impUnsupportedHWIntrinsic(CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* sig, bool mustExpand)
+{
+    // When we aren't supported we need to return nullptr when attempting to expand inline
+    // (gtIsRecursiveCall == false) as the gtNewMustThrowException node can mess with the
+    // inliner in unexpected ways.
```

Comment:
The comment looks a bit unclear.

Reply:
I'll try and clarify.

Basically, this is saying we've hit a failure case and we need to return `nullptr` if `mustExpand` is false, since that means we were attempting to expand the intrinsic inline another method. However, when `mustExpand` is true, it means we are expanding the intrinsic inline a GT_CALL to ourselves, so we can return the exception.

We do this because the inliner expects the intrinsic, not an exception, and will otherwise fail.

Reply:
Fixed. Hopefully the new wording is easier to understand.